src/main/java/redis/clients/jedis/RedisCredentialsProvider.java

```diff
@@ -0,0 +1,10 @@
+package redis.clients.jedis;
+
+import java.util.function.Supplier;
+
+public interface RedisCredentialsProvider extends Supplier<RedisCredentials> {
+
+  void prepare();
```

Comment:
Probably fine to have both methods as `default`. Also, you need to provide documentation on both methods what these should do and what the library expectation is that can happen or must not happen.